homeassistant/components/group/notify.py

```diff
@@ -29,46 +32,50 @@
 )
 
 
-def update(input_dict, update_source):
+def update(input_dict: dict[str, Any], update_source: dict[str, Any]) -> dict[str, Any]:
     """Deep update a dictionary.
 
     Async friendly.
     """
     for key, val in update_source.items():
         if isinstance(val, Mapping):
-            recurse = update(input_dict.get(key, {}), val)
+            recurse = update(input_dict.get(key, {}), val)  # type: ignore[arg-type]
             input_dict[key] = recurse
         else:
             input_dict[key] = update_source[key]
     return input_dict
 
 
-async def async_get_service(hass, config, discovery_info=None):
+async def async_get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> GroupNotifyPlatform:
     """Get the Group notification service."""
-    return GroupNotifyPlatform(hass, config.get(CONF_SERVICES))
+    return GroupNotifyPlatform(hass, config[CONF_SERVICES])
 
 
 class GroupNotifyPlatform(BaseNotificationService):
     """Implement the notification service for the group notify platform."""
 
-    def __init__(self, hass, entities):
+    def __init__(self, hass: HomeAssistant, entities: list[dict[str, Any]]) -> None:
         """Initialize the service."""
         self.hass = hass
         self.entities = entities
 
-    async def async_send_message(self, message="", **kwargs):
+    async def async_send_message(self, message: str = "", **kwargs: Any) -> None:
         """Send message to all entities in the group."""
         payload = {ATTR_MESSAGE: message}
```

Comment:
```suggestion
        payload: dict[str, Any] = {ATTR_MESSAGE: message}
```
Without the annotation, the generic is determined by the first assignment. I.e. it would be `dict[str, str]` instead.